cmake/X3D_MPI.cmake

```diff
@@ -0,0 +1,38 @@
+# MPI CMakeLists
+
+find_package(MPI REQUIRED)
+# Stop if there is no MPI_Fortran_Compiler
+if (MPI_Fortran_COMPILER)
+    message(STATUS "MPI_Fortran_COMPILER found: ${MPI_Fortran_COMPILER}")
+    message(STATUS "MPI_VERSION found: ${MPI_VERSION}")
+    # Try to guess the MPI type to adapt compilation flags if necessary
+    string(FIND "${MPI_Fortran_COMPILER}" "mpich" pos)
+    if(pos GREATER_EQUAL "0")
+      set(FIND_MPICH TRUE)
+      message(STATUS "MPI is MPICH type")
+    endif()
+    string(FIND "${MPI_Fortran_COMPILER}" "openmpi" pos)
+    if(pos GREATER_EQUAL "0")
+      set(FIND_OMPI TRUE)
+      message(STATUS "MPI is openMPI type")
+    endif()
+else (MPI_Fortran_COMPILER)
+    message(SEND_ERROR "This application cannot compile without MPI")
+endif(MPI_Fortran_COMPILER)
+# Warning if Include are not found => can be fixed with more recent cmake version
+if (MPI_FOUND)
+  message(STATUS "MPI FOUND: ${MPI_FOUND}")
+  include_directories(SYSTEM ${MPI_INCLUDE_PATH})
+  message(STATUS "MPI INCL ALSO FOUND: ${MPI_INCLUDE_PATH}")
+  # Force the mpirun to be coherent with the mpifortran
+  string(REGEX REPLACE "mpif90" "mpirun" PATH_TO_MPIRUN "${MPI_Fortran_COMPILER}")
+  string(REPLACE "mpiifort" "mpirun" PATH_TO_MPIRUN "${PATH_TO_MPIRUN}")
+  message(STATUS "Path to mpirun ${PATH_TO_MPIRUN}")
+  set(MPIEXEC_EXECUTABLE "${PATH_TO_MPIRUN}" CACHE STRING
+      "Force MPIRUN to be consistent with MPI_Fortran_COMPILER" FORCE)
```

Comment:
Is this where we go wrong with `srun` @rfj82982 ?

Reply:
Yes I have the solution to `srun`. I need to add the flags. I just don't understand why it is hanging the ctest.

Reply:
Ok now this should be sorted. CMake is enforcing the mpif90-mpirun consistency only at first iteration and there is bit in known issues about finding the mpi executable.